src/utils/generate_init_data.ts

```diff
@@ -0,0 +1,61 @@
+import { itole4, itobe4, itole2, concat } from "./byte_parsing";
+import { strToUtf8, strToUtf16LE, hexToBytes } from "./string_parsing";
+
+/**
+ * Generate the "cenc" init data for playready from the PlayreadyHeader string.
+ * @param {string} playreadyHeader - String representing the PlayreadyHeader XML.
+ * @returns {Uint8Array} The init data generated for that PlayreadyHeader.
+ * @see https://learn.microsoft.com/en-us/playready/specifications/playready-header-specification
+ */
+export function generatePlayReadyInitData(playreadyHeader: string): Uint8Array {
```

Comment:
I like that both of those functions are readable despite the fact that they are implementing technical details (with moreover the usual mixed endianness and repeating metadata for added fun :p).

However I'm now wondering if a function like this containing CDM implementation details isn't more at its place in the compat directory.

I'm legitimately wondering, it's a grey area for me. Thoughts?

Reply:
I think compat directory is for polyfill and device specific fixes. To me that does not include cdm specifications.

There is files like `src/main_thread/decrypt/utils/get_drm_system_id.ts` that are related to 
CDM details and they are not under the compat directory.
We can think of having a folder cdmProviders to put all the code related to specific CDMs but I don't know if there will be enough code to put in there.

Reply:
We could argue that this is in some way a device-specific fix

I feel like `compat` adds visibility to device-specific bugs and work-arounds. 
The advantage of isolating PlayReady-specific code to the `compat` directory would be to have on one side `decrypt` for the implementation of the EME specification logic and on the other `compat` for cdm-specific work-arounds which we could regularly re-check or make easier to generalize if the issue appears elsewhere (for example if we wanted to perform the same test with nagra, it could make sense to have a general `generateTestingInitData` function in `compat`).

However by adding a code round-trip to `compat` here, we complexify readability of the code a little.